source/extensions/filters/http/compressor/factory_base.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/server/filter_config.h"
+
+#include "extensions/filters/http/compressor/compressor_factory.h"
+#include "extensions/filters/http/compressor/compressor_library_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Compressor {
+
+template <class ConfigProto>
+class CompressorLibraryFactoryBase : public NamedCompressorLibraryConfigFactory {
```

Comment:
I am not quite seeing why we need the intermediary `NamedCompressorLibraryConfigFactory` class. I think we can just have the templated class derive directly from `Config::TypedFactory`